c_binding/utils.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Gris Ge <fge@redhat.com>
+ */
+
+#include "utils.h"
+
+#include <stdarg.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+
+int _check_null_ptr(char *err_msg, int arg_count, ...)
+{
+    int rc = LSM_ERR_OK;
+    va_list arg;
+    int i = 0;
+    void *ptr = NULL;
+
+    va_start(arg, arg_count);
+
+    for (; i < arg_count; ++i) {
+        ptr = va_arg(arg, void*);
+        if (ptr == NULL) {
+            _lsm_err_msg_set(err_msg, "Got NULL pointer in arguments");
+            rc = LSM_ERR_INVALID_ARGUMENT;
+            goto out;
+        }
+    }
+
+ out:
+    va_end(arg);
+    return rc;
+}
+
+void _be_raw_to_hex(uint8_t *raw, size_t len, char *out)
+{
+    size_t i = 0;
+
+    for (; i < len; ++i) {
+        snprintf(out + (i * 2), 3, "%02x", raw[i]);
+    }
+    out[len * 2] = 0;
+
+}
+
+bool _file_is_exist(const char *path)
```

Comment:
Rename to _file_exists? Reads a bit better in an if statement:

if (_file_exists("/dev/sg1"))
//blah